apps/03TicTacToe.js

```diff
@@ -24,23 +24,47 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
+  if ((board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn) || (board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn) || (board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn)) {
+    return true;
+  }
 }
 
 function verticalWin() {
-  // Your code here
+  if ((board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn) || (board[0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn) || (board[0][2] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn)) {
+    return true;
+  }
 }
 
 function diagonalWin() {
-  // Your code here
+  if ((board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn) || (board[0][2] === playerTurn && board[2][2] === playerTurn && board[2][0] === playerTurn)) {
+    return true;
+  }
 }
 
 function checkForWin() {
-  // Your code here
+  if (horizontalWin(true) || verticalWin(true) || diagonalWin(true)) {
+    console.log('Player ' + playerTurn + ' wins!');
+    return true;
+  }
+  return false;
+}
+
+function isSpaceAvailable(row, column) {
+  return (board[row][column] === "X" || board[row][column] === "O");
```

Comment:
This is confusing because the function name doesn't match what's returned. In other words, if there is an X or an O in the space, it returns `true`. But that means the space is unavailable, and should return `false`.